src/aviary/env.py

```diff
@@ -128,6 +129,21 @@ async def reset(self) -> tuple[Messages, list[Tool]]:
             Two-tuple of initial observations and tools.
         """
 
+    async def get_id(self) -> str | UUID:
+        """
+        Get a unique identifier for this environment.
+
+        The main use case is something like the ID of the task from a dataset.
+
+        The return should not be affected by state, in other words across reset/step,
+        the return value should not change.
+
+        This method is asynchronous to allow for DB transactions.
+        """
+        raise NotImplementedError(
+            f"Getting ID is not yet implemented for environment {type(self).__name__}."
+        )
+
```

Comment:
Can we have a sensible default, rather than blowing up?

Reply:
Yeah we can just do something like `return uuid.uuid4()` or `return str(id(self))`

I felt though that injecting a default here is a bit dangerous, so I made this new feature opt-in

Reply:
So upon reflection, IDs are driven by the dataset, imo the `Environment` should not prescribe a failover ID. It would be like having a middleware inject information, which decreases traceability

Reply:
Oh I was thinking `UNSET_ID = "UNSET ID"`, but I'm ok with this behavior too. 

And re: `str | None`/`is None`, I was referring to code like this: https://github.com/Future-House/aviary/blob/f68051997599ef07016b5a90865834cc03371b41/packages/gsm8k/src/aviary/envs/gsm8k/env.py#L207-L208 . The return value of `get_id` is always `str`, but seems like multiple envs use `None` as an internal sentinel for a missing ID.